### ISSUE TITLE
Minimized ol.js has a bug caused by minification

### ISSUE THREAD

**Issue by joux3** [openlayers/openlayers#8540]
I have seen the following error happen sometimes with the OpenLayers 5.1.3 distributed minified JS: `v is not a function @ ol.js:1:106127`.

The corresponding unminified code is at: https://github.com/openlayers/openlayers/blob/2342ec90aaca15fd222d37e65272015f789dc506/src/ol/View.js#L588-L595

The minified code is (formatted):
```
if (o) {
    t.animations_[r] = null, t.setHint(rs.ANIMATING, -1);
    var v = n[0].callback;
    v && setTimeout(function() {
        v(!0)
    }, 0)
}
```

Now the difference might not seem like much but there is a small difference. ES6 `const callback` is a block level declaration while `var v` gets hoisted to the function level. Usually this is a very small nuance but this time it causes a difference in functionality. This is because the `var` is used multiple times in a `for` loop where latter executions of the loop overwrite the same variable used by previous executions. See the sample test case at the bottom of this post for the difference.

The original version also has a call to `animationCallback` which would fix the difference between `const` and `var`, but it gets inlined by the minification process and therefore an extra closure containing the correct value for `callback` is not created.

The subtle difference between the behaviour of `const` and `var` causes that `callback` (`v` minified) can be falsy even with the if guard. I can not propose a good way to ensure that minification does not cause this error but somehow marking `animationCallback` as not inlinable should fix the bug.

Sorry for not having a repro case, but I can swear that I have seen this error happen many times. Luckily these have happened during a CI test case and not with something that's visible to users (I hope).

<details><summary>Sample test case between const and var</summary>

```
function varTest() {
  const input = [1,2,3]
  for (var i = 0; i < input.length; i++) {
    var number = input[i]
    setTimeout(() => {
      console.log(number)
    }, 0)
  }
}

function constTest() {
  const input = [1,2,3]
  for (var i = 0; i < input.length; i++) {
    const number = input[i]
    setTimeout(() => {
      console.log(number)
    }, 0)
  }
}

varTest() // writes 3, 3, 3
constTest() // writes 1, 2, 3
```

</details>

**Comment by ahocevar** [comment]
Duplicate of #8500, fixed already with #8503.

**Comment by joux3** [comment]
Shame on me for only searching for `v is not a function`. Thanks!
